eng/Build.ps1

```diff
@@ -148,8 +149,17 @@ function Process-Arguments() {
         $script:testVs = $True
     }
 
+    if ($testAllButIntegration) {
```

Comment:
That looks a bit clunky, we probably should just have a separate leg, which runs with `testIntegration` only.

Reply:
So alternatively we can just set the value to false [here](https://github.com/dotnet/fsharp/blob/main/eng/Build.ps1#LL147-L147C40) but that will make the name "-test**All**" a lie. I wouldn't rename it to "-testAllUnit" either as, well, not many of those tests are unit tests. Hence this new switch... But I don't have a very strong stance on this, just making a point.

Reply:
I'm okay with the test framework having explicit subsets of tests.  We do it a lot for main test suites.  I.e. testCambridge, --testFSharpQa etc.